### PR TITLE
#29 Create a method to return RPC server on the top block view

### PR DESCRIPTION
**Pull request by THEozmic**
## Description
Created `getRpcServer` action, updated the `topBlocks` reducer and displayed the rpc server on the 
view

## Related Issue
https://github.com/DAVFoundation/xplore/issues/29

## Motivation and Context
Helps user see what their RPC server is

## How Has This Been Tested?
- Wrote a new test case
- Ran `npm test`
- Looked through the app and made sure everything still works as expected

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test/specs/reducers/transactions.spec.js [hunk 1]
```diff
@@ -7,9 +7,9 @@ describe('Transactions reducer',() =>{
       const action = {
         type: 'GET_TRANSACTIONS_FULFILLED',
         payload: [{ hash: '0x9fc76417374aa880d4449a1f7f31ec597f00b1f6f3dd2d66f4c9c6c445836d8b', 
-                    from: '0xa94f5374fce5edbc8e2a8697c15331677e6ebf0b',
-                    to: '0x6295ee1b4f6dd65047762f924ecd367c17eabf8f',
-                    blockNumber: '3' }],
+          from: '0xa94f5374fce5edbc8e2a8697c15331677e6ebf0b',
+          to: '0x6295ee1b4f6dd65047762f924ecd367c17eabf8f',
+          blockNumber: '3' }],
```
Comment (griffobeid) [inline test/specs/reducers/transactions.spec.js]: The changes here are irrelevant to the issue and should be made in a separate pull request.

File: test/specs/reducers/topBlocks.spec.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import reducer from '../../../src/reducers/topBlocks';
+
+describe('Top Block reducer', () => {
+  describe('When GET_RPC_SERVER action is fired', () => {
```
Comment (griffobeid) [inline test/specs/reducers/topBlocks.spec.js]: Your description here should be consistent with what you end up naming the action.

File: src/containers/TopBlocksContainer.jsx [hunk 3]
```diff
@@ -1,8 +1,13 @@
...
+const mapDispatchToProps = (dispatch) => ({
+  getRpcServer: () => dispatch(getRpcServer())
+});
 
 const mapStateToProps = (state) => ({
   blockList: state.topBlocks.blockList
 });
 
-export default connect(mapStateToProps)(TopBlocks);
+export default connect(mapStateToProps, mapDispatchToProps)(TopBlocks);
```
Comment (griffobeid) [inline src/containers/TopBlocksContainer.jsx]: Correct me if I'm wrong here @TalAter, but I believe the changes to this file are unnecessary. You are making a call to `getRpcServer()` inside of the `topBlocks` reducer.

File: src/components/TopBlocks.jsx [hunk 4]
```diff
@@ -49,13 +48,15 @@ class BlockItem extends Component {
 }
 
 TopBlocks.propTypes = {
-  blockList: PropTypes.array
+  blockList: PropTypes.array,
+  getRpcServer: PropTypes.func
```
Comment (griffobeid) [inline src/components/TopBlocks.jsx]: What is the purpose line 52 here? The RPC server should already be fed by the `blockList` prop.

File: src/actions/index.js [hunk 5]
```diff
@@ -25,3 +25,5 @@ export const getMiningStatus = createAction('GET_MINING_STATUS', getMiningStatus
 export const getTransactionCount = createAction('GET_TRANSACTION_COUNT', getTransactionCountCall);
 
 export const getLatestBlock = createAction('GET_LATEST_BLOCK', getLatestBlockCall);
+
+export const getRpcServer = createAction('GET_TOP_BLOCKS', getRpcServerCall);
```
Comment (griffobeid) [inline src/actions/index.js]: I feel as though 'GET_TOP_BLOCKS' is not the most appropriate name for this action. Maybe go with 'GET_RPC_SERVER' instead.

**Review by griffobeid — CHANGES_REQUESTED** [review]
Overall, excellent pull request @THEozmic! I just have a few suggestions for you to make it even better.

**Comment by THEozmic** [comment]
Hey @griffobeid, I expected to get a few these comments because I wasn't very sure about how and where to place things. Thanks for the feedback, I'll make the changes as soon as is possible.

**Comment by THEozmic** [comment]
@griffobeid I've made some changes, please review and let me know if anything else isn't right.

**Comment by griffobeid** [comment]
Awesome, @THEozmic 💪 thanks for another great pull request! 😄

**Pull request merged**